pkg/operation/seed/seed.go

```diff
@@ -828,151 +828,351 @@ func BootstrapCluster(ctx context.Context, k8sGardenClient, k8sSeedClient kubern
 		return err
 	}
 
-	if err := deployGardenerResourceManager(ctx, k8sSeedClient, v1beta1constants.GardenNamespace, imageVector); err != nil {
-		return err
-	}
+	return runCreateSeedFlow(ctx, k8sSeedClient, k8sGardenClient, imageVector, imageVectorOverwrites, seed, seedLogger)
+}
 
-	// Deploy component specific resources
-	bootstrapComponents, err := bootstrapComponents(k8sSeedClient, v1beta1constants.GardenNamespace, imageVector, imageVectorOverwrites)
+func runCreateSeedFlow(ctx context.Context, sc, gc kubernetes.Interface, imageVector imagevector.ImageVector, imageVectorOverwrites map[string]string, seed *Seed, seedLogger *logrus.Entry) error {
+	var (
+		g         = flow.NewGraph("Seed cluster creation")
+		namespace = v1beta1constants.GardenNamespace
+	)
+	kubernetesVersion, err := semver.NewVersion(sc.Version())
 	if err != nil {
 		return err
 	}
 
-	var bootstrapFunctions []flow.TaskFn
-	for _, componentFn := range bootstrapComponents {
-		fn := componentFn
-		bootstrapFunctions = append(bootstrapFunctions, func(ctx context.Context) error {
-			return component.OpWaiter(fn).Deploy(ctx)
-		})
+	if err := updateDNSProviderSecret(ctx, sc, gc, seed); err != nil {
+		return err
 	}
 
-	return flow.Parallel(bootstrapFunctions...)(ctx)
-}
+	// setup for flow graph
+	var dnsEntry component.DeployWaiter
+	dnsConfig := seed.Info.Spec.DNS
+	if managedIngress(seed) {
+		ingressLoadBalancerAddress, err := getManagedIngressLoadBalancerAddress(ctx, sc, seedLogger)
+		if err != nil {
+			return err
+		}
+		dnsEntry = getManagedIngressDNSEntry(sc, seed.GetIngressFQDN("*"), ingressLoadBalancerAddress, seedLogger)
+	} else {
+		dnsEntry = getManagedIngressDNSEntry(sc, seed.GetIngressFQDN("*"), "", seedLogger)
+	}
 
-// DebootstrapCluster deletes certain resources from the seed cluster.
-func DebootstrapCluster(ctx context.Context, k8sSeedClient kubernetes.Interface) error {
-	bootstrapComponents, err := bootstrapComponents(k8sSeedClient, v1beta1constants.GardenNamespace, nil, nil)
+	grmImage, err := imageVector.FindImage(charts.ImageNameGardenerResourceManager, imagevector.RuntimeVersion(sc.Version()), imagevector.TargetVersion(sc.Version()))
 	if err != nil {
 		return err
 	}
-
-	// Delete component specific resources
-	var debootstrapFunctions []flow.TaskFn
-	for _, componentFn := range bootstrapComponents {
-		fn := componentFn
-		debootstrapFunctions = append(debootstrapFunctions, func(ctx context.Context) error {
-			return component.OpDestroyAndWait(fn).Destroy(ctx)
-		})
+	cfg := resourcemanager.Values{
+		ConcurrentSyncs:  pointer.Int32Ptr(20),
+		HealthSyncPeriod: utils.DurationPtr(time.Minute),
+		ResourceClass:    pointer.StringPtr(v1beta1constants.SeedResourceManagerClass),
+		SyncPeriod:       utils.DurationPtr(time.Hour),
 	}
+	rm := resourcemanager.New(sc.Client(), namespace, grmImage.String(), 1, cfg)
```

Comment:
Personal preference:

```suggestion
	resourceManager := resourcemanager.New(sc.Client(), namespace, grmImage.String(), 1, resourcemanager.Values{
		ConcurrentSyncs:  pointer.Int32Ptr(20),
		HealthSyncPeriod: utils.DurationPtr(time.Minute),
		ResourceClass:    pointer.StringPtr(v1beta1constants.SeedResourceManagerClass),
		SyncPeriod:       utils.DurationPtr(time.Hour),
	})
```